src/table/src/EditableCell.js

```diff
@@ -18,82 +19,100 @@ const EditableCell = memo(function EditableCell(props) {
     isSelectable = true,
     textProps = emptyProps,
     autoFocus = false,
+    onChange,
     ...rest
   } = props
 
   let cursor = 'text'
 
-  const [mainRef, setMainRef] = useState()
+  const mainRef = useRef(null)
   const [value, setValue] = useState(children)
   const [isEditing, setIsEditing] = useState(autoFocus)
+  const onChangeRef = useLatest(onChange)
 
   useEffect(() => {
     setValue(children)
   }, [children])
 
-  const handleDoubleClick = () => {
+  const handleDoubleClick = useCallback(() => {
     if (disabled || !isSelectable) return
 
     setIsEditing(true)
-  }
+  }, [disabled, isSelectable])
+
+  const handleKeyDown = useCallback(
+    e => {
+      if (disabled) return
+      const { key } = e
+
+      /**
+       * When the user presses a character on the keyboard, use that character
+       * as the value in the text field.
+       */
+      if (key === 'Enter' || key === 'Shift') {
+        setIsEditing(true)
+      } else if (key.match(/^[a-z]{0,10}$/) && !e.metaKey && !e.ctrlKey && !e.altKey) {
+        setIsEditing(true)
+        setValue(prev => prev + key)
+      }
+    },
+    [disabled]
+  )
 
-  const handleKeyDown = e => {
-    if (disabled) return
-    const { key } = e
-
-    /**
-     * When the user presses a character on the keyboard, use that character
-     * as the value in the text field.
-     */
-    if (key === 'Enter' || key === 'Shift') {
-      setIsEditing(true)
-    } else if (key.match(/^[a-z]{0,10}$/) && !e.metaKey && !e.ctrlKey && !e.altKey) {
-      setIsEditing(true)
-      setValue(value + key)
-    }
-  }
+  const handleFieldChangeComplete = useCallback(
+    value => {
+      setIsEditing(false)
+      setValue(value)
 
-  const handleFieldChangeComplete = value => {
-    const { onChange } = rest
+      safeInvoke(onChangeRef.current, value)
 
-    setIsEditing(false)
-    setValue(value)
+      if (mainRef.current && isSelectable) {
+        mainRef.current.focus()
+      }
+    },
+    // onChangeRef is a ref
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [isSelectable]
+  )
 
-    safeInvoke(onChange, value)
+  const handleFieldCancel = useCallback(() => {
+    setIsEditing(false)
+  }, [])
 
-    if (mainRef && isSelectable) {
-      mainRef.focus()
+  const handleClick = useCallback(() => {
+    if (mainRef.current) {
+      mainRef.current.focus()
     }
-  }
+  }, [])
 
-  const handleFieldCancel = () => {
-    setIsEditing(false)
-  }
-
-  const handleClick = () => {
-    if (mainRef) mainRef.focus()
-  }
+  const getTargetRef = useCallback(() => mainRef.current, [])
 
   if (disabled) {
     cursor = 'not-allowed'
   } else if (isSelectable) {
     cursor = 'default'
   }
 
+  const lessOpacity = useMemo(() => disabled || (!value && placeholder), [disabled, value, placeholder])
+
+  const mergedTextProps = useMemo(
+    () => ({
+      size,
+      opacity: lessOpacity ? 0.5 : 1,
+      ...textProps
+    }),
+    [lessOpacity, size, textProps]
```

Comment:
Debating on whether we should do `...textProps` but probably not given that this might change on a mounted component? Would it?

Reply:
i think even if the wrapped the change in a useMemo the object ref would be changing here given its a single incoming prop (vs say `const Example = ({otherProps, ...textProps}) => {}`

Reply:
It would, basically if the parent rerenders and they are passing `textProps={{}}` it'll change. but i can't guarantee that it has the same number of keys, therefore it is unsafe to spread as deps because it could change across renders.